wait-for-deployment

```diff
@@ -36,28 +37,48 @@ get_deployment_jsonpath() {
   kubectl get deployment "${deployment}" -o "jsonpath=${_jsonpath}"
 }
 
+get_statuses_for_deployment() {
+  kubectl get pods --selector=app=${deployment} -o 'jsonpath={.items[*].status.conditions[*].type}'
+}
+
+count_ready_pods() {
+  get_statuses_for_deployment | tr ' ' '\n' | grep Ready | wc -l
+}
+
 if [[ $# != 1 ]]; then
-  echo "usage: $(basename $0) <deployment>" >&2
+  echo "Usage: $(basename $0) <deployment>" >&2
   exit 1
 fi
 
 readonly deployment="$1"
 
 readonly generation=$(get_generation)
-echo "waiting for specified generation ${generation} to be observed"
-while [[ $(get_observed_generation) -lt ${generation} ]]; do
+echo "Waiting for specified generation: ${generation} for deployment ${deployment} to be observed"
```

Comment:
How about having a similar status output as with the generations below; that is, have this line say something like

`Expected generation for deployment ${deployment}: ${generation}`

...